admin-dev/themes/new-theme/js/components/modal/progress-modal.ts

```diff
@@ -0,0 +1,421 @@
+    this.switchButton.classList.add(
+      'switch-to-progress-button',
+      'btn',
+      'btn-secondary'
+    );
+    this.switchButton.innerHTML = 'Back to processing';
+
+    this.downloadErrorsButton = document.createElement('div');
+    this.downloadErrorsButton.classList.add(
+      'download-error-log',
+      'btn',
+      'btn-secondary'
+    );
+    this.downloadErrorsButton.innerHTML = 'Download error log';
+
+    this.footer.append(this.switchButton);
+    this.footer.append(this.downloadErrorsButton);
+    this.body.append(this.footer);
+
+
+    // Modal title element
+    if (params.modalTitle) {
+      this.title = document.createElement('h4');
+      this.title.classList.add('modal-title');
+      this.title.innerHTML = params.modalTitle;
+    }
+
+    // Constructing the modal
+    if (this.title) {
+      this.header.appendChild(this.title);
+    }
+    this.content.append(this.header, this.body);
+  }
+}
+
+/**
+ * ConfirmModal component
+ *
+ * @param {InputConfirmModalParams} params
+ * @param {Function} confirmCallback @deprecated You should rely on the confirmCallback param
+ * @param {Function} cancelCallback @deprecated You should rely on the closeCallback param
+ */
+export class ProgressModal extends Modal implements ProgressModalType {
+  modal!: ProgressModalParentContainerType;
+  doneCount !: number;
+  total !: number;
+  errors !: Array<string>;
+  cancelCallback !: () => void;
+  progressModal !: ProgressModalContainer;
+  errorModal !: ProgressModalErrorContainer;
+  constructor(
+    inputParams: InputProgressModalParams,
+    total: number,
+    confirmCallback: (event: Event) => void,
+    cancelCallback: () => void,
+  ) {
+    const params: ProgressModalParams = {
+      id: 'progress-modal',
+      customButtons: [],
+      closable: false,
+      modalTitle: "Progress action",
+      total: total,
+      dialogStyle: {},
+      confirmCallback,
+      ...inputParams,
+    };
+
+    super(params);
+    this.cancelCallback = cancelCallback;
+    this.total = total;
+    this.errors = [];
+  }
+
+  protected initContainer(params: ProgressModalParams): void {
+    this.modal = new ProgressModalParentContainer(params);
+    super.initContainer(params);
+    this.progressModal = new ProgressModalContainer(params);
+    this.errorModal = new ProgressModalErrorContainer(params);
+
+    this.modal.dialog.appendChild(this.progressModal.content);
+
+    this.initListeners();
+  }
+
+  public modalActionSuccess(doneCount: number)
+  {
+    this.updatePercentage(doneCount);
+  }
+
+  public modalActionError(doneCount: number, error: string)
+  {
+    this.updatePercentage(doneCount);
+    this.addError(error);
+  }
+
+  private updatePercentage(doneCount: number)
+  {
+    let progressBarDone = doneCount * 100 / this.total;
+    this.progressModal.progressDone.style.width = progressBarDone+'%';
+    this.progressModal.progressPercentage.innerHTML = progressBarDone+'%';
+  }
+
+  private addError(error: string)
```

Comment:
```suggestion
 addError(error: string)
```

Reply:
Oh I see what you mean. Separated the methods.